simple-loadflow/src/main/java/com/powsybl/loadflow/simple/equations/LoadFlowMatrix.java

```diff
@@ -71,105 +72,155 @@ private static Bus getBus2(Branch b) {
+    public static double[] buildDcRhs(IndexedNetwork network, int slackBusNum) {
+        double[] rhs = new double[network.getBusCount()];
 
         for (Generator gen : network.get().getGenerators()) {
             Bus bus = gen.getTerminal().getBusView().getBus();
+            if (bus == null || !bus.isInMainConnectedComponent()) {
+                continue;
+            }
             int num = network.getIndex(bus);
-            if (num == 0) {
+            if (num == slackBusNum) {
                 continue;
             }
-            rhs.set(num, 0, gen.getTargetP());
+            rhs[num] -= gen.getTargetP();
         }
 
         for (Load load : network.get().getLoads()) {
             Bus bus = load.getTerminal().getBusView().getBus();
+            if (bus == null || !bus.isInMainConnectedComponent()) {
+                continue;
+            }
             int num = network.getIndex(bus);
-            if (num == 0) {
+            if (num == slackBusNum) {
                 continue;
             }
-            rhs.set(num, 0, load.getP0());
+            rhs[num] += load.getP0();
+        }
+
+        for (HvdcLine line : network.get().getHvdcLines()) {
+            Bus bus1 = line.getConverterStation1().getTerminal().getBusView().getBus();
+            Bus bus2 = line.getConverterStation2().getTerminal().getBusView().getBus();
+            double p = line.getConvertersMode() == HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER
+                    ? line.getActivePowerSetpoint()
+                    : -line.getActivePowerSetpoint();
+            if (bus1 != null && bus1.isInMainConnectedComponent()) {
+                int num = network.getIndex(bus1);
+                if (num != slackBusNum) {
+                    rhs[num] += p;
+                }
+            }
+            if (bus2 != null && bus2.isInMainConnectedComponent()) {
+                int num = network.getIndex(bus2);
+                if (num != slackBusNum) {
+                    rhs[num] -= p;
+                }
+            }
         }
 
         return rhs;
     }
 
-    public static void updateNetwork(IndexedNetwork network, MatrixStore<Double> lhs) {
+    public static void updateDcNetwork(IndexedNetwork network, double[] lhs) {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        for (Bus bus : network.get().getBusView().getBuses()) {
+            bus.setV(Double.NaN);
+            bus.setAngle(Double.NaN);
+        }
+
         for (Bus bus : network.getBuses()) {
             int num = network.getIndex(bus);
-            bus.setAngle(lhs.get(num, 0));
+            bus.setV(bus.getVoltageLevel().getNominalV());
+            bus.setAngle(Math.toDegrees(lhs[num]));
         }
 
         for (Branch branch : network.get().getBranches()) {
             DcFlowEquations eq = DcFlowEquations.of(branch);
             branch.getTerminal1().setP(eq.p1());
+            branch.getTerminal1().setQ(Double.NaN);
```

Comment:
We should tell in the javadoc that a side effect of the DC computation is Q = NaN.
(which means I = NaN too I think)